luigi/worker.py

```diff
@@ -102,11 +102,14 @@ def __init__(self, task, worker_id, result_queue, random_seed=False, worker_time
         self.timeout_time = time.time() + worker_timeout if worker_timeout else None
 
     def _run_get_new_deps(self):
+        run_again = False
         try:
             task_gen = self.task.run(tracking_url_callback=self.tracking_url_callback)
         except TypeError as ex:
             if 'unexpected keyword argument' not in getattr(ex, 'message', ex.args[0]):
```

Comment:
It was the best way we were able to implement #1298.

I could send a PR later that adds comments around the code if it helps.